aether-kernel/aether/kernel/api/utils.py

```diff
@@ -52,3 +56,125 @@ def merge_objects(source, target, direction):
     else:
         result = target
     return result
+
+
+def get_unique_schemas_used(mappings_ids):
+    result = {}
+    schemas = models.Schema.objects.filter(schemadecorators__mappings__id__in=mappings_ids)
+    for schema in schemas:
+        other_linked_mappings = models.Mapping.objects.filter(
+            schemadecorators__schema__id=schema.id
+        ).exclude(id__in=mappings_ids)
```

Comment:
What happens when the schema is not used in any mapping but in the entities?

Reply:
The purpose of this for now is to identify delete-able schemas from the perspective of mappings. If it isn't used in a mapping, it will not be returned as the filter is on mappings

Reply:
Let me explain it better... what happens in this case:

1. Schema `S` is used but mappings `M` and `N`, and entities `E` (generated by `M`) and `F` (generated by `N`).
2. User `A` decides to delete mapping `N` (no more entities need to be produced with this mapping) but not the linked entities, so entity `F` is now _"mapping-orphan"_ but still belongs to `S`.
3. User `B` decides to check mapping `M` and the schemas linked to it.

With your approach schema `S` is only used by `M` (`result['S']['is_unique'] = True`) so user `B` decides to delete also the schema and the `schema.delete()` not only allows you to do that it also deletes entity `F` that user `A` wanted to keep.

Reply:
Yes, I understand the issue. Which is why I brought up the suggestion sometime back about `Entity` not having cascade deletes at all.

Reply:
The question now, is should schema's be prevented from being deleted if entities outside the delete scope utilise them?

Reply:
Do we want to keep Entities without the schema that generated them? @adamvert 
Do we want orphan data? :scream:

Reply:
According to the [spec](https://jira.ehealthafrica.org/browse/AET-476?focusedCommentId=36853&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-36853), deleting a Schema or Project Schema will _always_ delete the Entities

Reply:
I don't see that this is resolved. :(

Reply:
I would love to have at least a comment here with the risks of flagging a schema with entities but not mappings as "unique"